runtime/src/mm/cpp/ThreadState.cpp

```diff
@@ -35,26 +34,26 @@ ALWAYS_INLINE mm::ThreadState mm::SwitchThreadState(ThreadData* threadData, Thre
     return oldState;
 }
 
-ALWAYS_INLINE void mm::AssertThreadState(ThreadData* threadData, ThreadState expected) noexcept {
+ALWAYS_INLINE ThreadState kotlin::SwitchThreadState(MemoryState* thread, ThreadState newState) noexcept {
+    return SwitchThreadState(thread->GetThreadData(), newState);
+}
+
+ALWAYS_INLINE void kotlin::AssertThreadState(mm::ThreadData* threadData, ThreadState expected) noexcept {
     auto actual = threadData->state();
     RuntimeAssert(actual == expected,
                   "Unexpected thread state. Expected: %s. Actual: %s.",
                   stateToString(expected), stateToString(actual));
 }
 
-mm::ThreadStateGuard::ThreadStateGuard(ThreadData* threadData, ThreadState state) noexcept : threadData_(threadData) {
-    oldState_ = SwitchThreadState(threadData, state);
-}
-
-mm::ThreadStateGuard::~ThreadStateGuard() noexcept {
-    SwitchThreadState(threadData_, oldState_);
+ALWAYS_INLINE void kotlin::AssertThreadState(MemoryState* thread, ThreadState expected) noexcept {
+    AssertThreadState(thread->GetThreadData(), expected);
 }
 
 extern "C" ALWAYS_INLINE RUNTIME_NOTHROW void Kotlin_mm_switchThreadStateNative() {
```

Comment:
What do you think about moving these two inside `Memory.cpp`? That's the place riddled with `extern "C"` stuff anyway, and pretty much serving as the API surface for the module.

Reply:
Sounds reasonable. Moved.